### PR TITLE
Naming convention for Vulkan

### PR DESCRIPTION
**Pull request by Maksasj**
Closing #97

### CODE REVIEW

File: src/bebone/gfx/vulkan/vulkan_pipeline_manager.h [hunk 1]
```diff
@@ -20,12 +20,12 @@ namespace bebone::gfx::vulkan {
 
             VulkanManagedPipelineTuple create_pipeline(
                 std::shared_ptr<VulkanDevice>& device,
-                std::shared_ptr<VulkanSwapChain>& swapChain,
+                const std::shared_ptr<VulkanSwapChain>& swapChain,
```
Comment (Soskar1) [inline src/bebone/gfx/vulkan/vulkan_pipeline_manager.h:23]: swapChain -> swap_chain

File: src/bebone/gfx/vulkan/vulkan_pipeline_manager.h [hunk 2]
```diff
@@ -20,12 +20,12 @@ namespace bebone::gfx::vulkan {
...
                 const std::string& vertex_shader_file_path,
                 const std::string& fragment_shader_file_path,
                 const std::vector<VulkanConstRange>& constantRanges,
```
Comment (Soskar1) [inline src/bebone/gfx/vulkan/vulkan_pipeline_manager.h:26]: constantRanges -> constant_ranges

**Review by Soskar1 — CHANGES_REQUESTED** [review]
Fix everything

**Comment by Soskar1** [comment]
[bufferInfo -> buffer_info](https://github.com/Maksasj/bebone/blob/97-naming-convention-for-vulkan/src/bebone/gfx/vulkan/vulkan_buffer.h#L53)

**Comment by Soskar1** [comment]
[const std::vector<std::shared_ptr<VulkanDescriptorSet>> & descriptors -> const std::vector<std::shared_ptr<VulkanDescriptorSet>>& descriptors](https://github.com/Maksasj/bebone/blob/ca97250b5c76d21a89b021381ffc38bb15d98fa7/src/bebone/gfx/vulkan/vulkan_pipeline_tuples.cpp#L12)

**Comment by Soskar1** [comment]
[VulkanDevice &device -> VulkanDevice& device](https://github.com/Maksasj/bebone/blob/ca97250b5c76d21a89b021381ffc38bb15d98fa7/src/bebone/gfx/vulkan/vulkan_pipeline_tuples.cpp#L55)

**Comment by Soskar1** [comment]
[const std::vector<std::shared_ptr<VulkanDescriptorSet>> & descriptors -> const std::vector<std::shared_ptr<VulkanDescriptorSet>>& descriptors](https://github.com/Maksasj/bebone/blob/ca97250b5c76d21a89b021381ffc38bb15d98fa7/src/bebone/gfx/vulkan/vulkan_pipeline_tuples.h#L31)

**Comment by Soskar1** [comment]
[VulkanDevice &device -> VulkanDevice& device](https://github.com/Maksasj/bebone/blob/ca97250b5c76d21a89b021381ffc38bb15d98fa7/src/bebone/gfx/vulkan/vulkan_pipeline_tuples.h#L43)

File: src/bebone/gfx/vulkan/vulkan_buffer.cpp [hunk 3]
```diff
@@ -8,28 +8,29 @@ namespace bebone::gfx::vulkan {
...
-        createInfo.pNext = nullptr;
-        createInfo.flags = bufferInfo.flags;
-        createInfo.size = size;
-        createInfo.usage = bufferInfo.usage;
-        createInfo.sharingMode = bufferInfo.sharingMode;
-        createInfo.queueFamilyIndexCount = bufferInfo.queueFamilyIndexCount;
-        createInfo.pQueueFamilyIndices = bufferInfo.pQueueFamilyIndices;
-
-        if (vkCreateBuffer(device.device(), &createInfo, nullptr, &backend) != VK_SUCCESS) {
+        VkBufferCreateInfo create_Info{};
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_buffer.cpp:13]: ok my lord

File: src/bebone/gfx/vulkan/vulkan_descriptor_set_layout.cpp [hunk 4]
```diff
@@ -4,59 +4,64 @@
...
+        layout_info.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_CREATE_INFO;
+        layout_info.bindingCount = bindings.size();
+        layout_info.pBindings = bindings.data();
+        layout_info.flags = VK_DESCRIPTOR_SET_LAYOUT_CREATE_UPDATE_AFTER_BIND_POOL_BIT_EXT;
 
         VkDescriptorSetLayoutBindingFlagsCreateInfoEXT extendedInfo;
         extendedInfo.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_SET_LAYOUT_BINDING_FLAGS_CREATE_INFO_EXT;
         extendedInfo.pNext = nullptr;
-        extendedInfo.bindingCount = bindingFlags.size();
+        extendedInfo.bindingCount = binding_flags.size();
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_descriptor_set_layout.cpp:47]: sure my lord

File: src/bebone/gfx/vulkan/vulkan_instance.cpp [hunk 5]
```diff
@@ -7,40 +7,8 @@ namespace bebone::gfx::vulkan {
...
-        return true;
-    }
-
-    void VulkanInstance::createInstance() {
-        if (enable_validation_layers && !checkValidationLayerSupport()) {
+        if (enable_validation_layers && !check_validation_layer_support())
             throw std::runtime_error("validation layers requested, but not available!");
-        }
 
         VkApplicationInfo appInfo = {};
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_instance.cpp:13]: please blame me more 🥵

File: src/bebone/gfx/vulkan/vulkan_instance.cpp [hunk 6]
```diff
@@ -54,73 +22,98 @@ namespace bebone::gfx::vulkan {
         createInfo.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_instance.cpp:22]: ok

File: src/bebone/gfx/vulkan/vulkan_instance.cpp [hunk 7]
```diff
@@ -54,73 +22,98 @@ namespace bebone::gfx::vulkan {
...
         createInfo.pApplicationInfo = &appInfo;
 
-        auto extensions = getRequiredExtensions();
+        auto extensions = get_required_extensions();
         createInfo.enabledExtensionCount = static_cast<uint32_t>(extensions.size());
         createInfo.ppEnabledExtensionNames = extensions.data();
 
         if (enable_validation_layers) {
             VkDebugUtilsMessengerCreateInfoEXT debugCreateInfo;
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_instance.cpp:30]: ok sir

File: src/bebone/gfx/vulkan/vulkan_instance.cpp [hunk 8]
```diff
@@ -54,73 +22,98 @@ namespace bebone::gfx::vulkan {
...
+        uint32_t layer_count;
+        vkEnumerateInstanceLayerProperties(&layer_count, nullptr);
+
+        auto availableLayers = std::vector<VkLayerProperties>(layer_count);
+        vkEnumerateInstanceLayerProperties(&layer_count, availableLayers.data());
+
+        for(const char *layerName : validation_layers) {
+            bool layerFound = false;
+
+            for (const auto &layerProperties : availableLayers) {
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_instance.cpp:61]: ok

File: src/bebone/gfx/vulkan/vulkan_device.cpp [hunk 9]
```diff
@@ -370,101 +399,93 @@ namespace bebone::gfx::vulkan {
...
-            }
         }
 
         throw std::runtime_error("failed to find supported format!");
     }
 
-    uint32_t VulkanDevice::find_memory_type(uint32_t typeFilter, VkMemoryPropertyFlags properties) {
-        VkPhysicalDeviceMemoryProperties memProperties;
-        vkGetPhysicalDeviceMemoryProperties(physicalDevice, &memProperties);
+    uint32_t VulkanDevice::find_memory_type(uint32_t type_filter, VkMemoryPropertyFlags m_properties) {
```
[the remark this answers is not in the document]
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_device.cpp:475]: intended, stands for **memory properties**, but ok, will change to mem_properties

File: src/bebone/gfx/vulkan/vulkan_pipeline_manager.h (hunk 1 above)
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_pipeline_manager.h:23]: ok

File: src/bebone/gfx/vulkan/vulkan_pipeline_manager.h (hunk 2 above)
Comment (Maksasj) [reply src/bebone/gfx/vulkan/vulkan_pipeline_manager.h:26]: ok

**Review by Soskar1 — APPROVED** [review]

**Pull request merged**